lib/fileutils.rb

```diff
@@ -842,25 +853,70 @@ def copy_file(src, dest, preserve = false, dereference = true)
   end
   module_function :copy_file
 
-  #
-  # Copies stream +src+ to +dest+.
-  # +src+ must respond to #read(n) and
-  # +dest+ must respond to #write(str).
+  # Copies \IO stream +src+ to \IO stream +dest+ via
+  # {IO.copy_stream}[https://docs.ruby-lang.org/en/master/IO.html#method-c-copy_stream].
   #
   def copy_stream(src, dest)
     IO.copy_stream(src, dest)
   end
   module_function :copy_stream
 
+  # Moves files from +src+ to +dest+.
+  # If +src+ and +dest+ are on different devices,
+  # first copies, then removes +src+.
+  #
+  # If +src+ is the path to a single file or directory and +dest+ does not exist,
+  # moves +src+ to +dest+:
+  #
+  #   system('tree --charset=ascii src0')
+  #   src0
+  #   |-- src0.txt
+  #   `-- src1.txt
+  #   File.exist?('dest0') # => false
+  #   FileUtils.mv('src0', 'dest0')
+  #   File.exist?('src0')  # # => false
+  #   system('tree --charset=ascii dest0')
+  #   dest0
+  #   |-- src0.txt
+  #   `-- src1.txt
+  #
+  # If +src+ is an array of paths to files and directories
+  # and +dest+ is the path to a directory,
+  # copies from each path in the array to th
+  #
+  #   File.file?('src1.txt') # => true
```

Comment:
Having `src1.txt` and `src1/src1.txt` is very confusing.  I originally thought the documentation was incorrect due to this.  Please rename `src1.txt` to `src0.txt` or `src2.txt` to prevent this confusion.